internal/uvm/computeagent_test.go

```diff
@@ -0,0 +1,229 @@
+package uvm
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Microsoft/hcsshim/internal/computeagent"
+	hcsschema "github.com/Microsoft/hcsshim/internal/hcs/schema2"
+	"github.com/Microsoft/hcsshim/internal/hns"
+)
+
+type testUtilityVM struct{}
+
+func (t *testUtilityVM) AddEndpointToNSWithID(ctx context.Context, nsID, nicID string, endpoint *hns.HNSEndpoint) error {
+	return nil
+}
+
+func (t *testUtilityVM) RemoveEndpointFromNS(ctx context.Context, id string, endpoint *hns.HNSEndpoint) error {
+	return nil
+}
+
+func (t *testUtilityVM) UpdateNIC(ctx context.Context, id string, settings *hcsschema.NetworkAdapter) error {
+	return nil
+}
+
+func TestAddNIC(t *testing.T) {
+	ctx := context.Background()
+
+	agent := &computeAgent{
+		uvm: &testUtilityVM{},
+	}
+
+	hnsGetHNSEndpointByName = func(endpointName string) (*hns.HNSEndpoint, error) {
+		return &hns.HNSEndpoint{
+			Namespace: &hns.Namespace{ID: "test-namespace-ID"},
+		}, nil
+	}
+
+	var (
+		testNICID        = "test-NIC-ID"
+		testEndpointName = "test-endpoint-name"
+	)
+
+	type config struct {
+		name          string
+		nicID         string
+		endpointName  string
+		errorExpected bool
+	}
+	tests := []config{
+		{
+			name:          "AddNIC returns no error",
+			nicID:         testNICID,
+			endpointName:  testEndpointName,
+			errorExpected: false,
+		},
+		{
+			name:          "AddNIC returns error with blank nic ID",
+			nicID:         "",
+			endpointName:  testEndpointName,
+			errorExpected: true,
+		},
+		{
+			name:          "AddNIC returns error with blank endpoint name",
+			nicID:         testNICID,
+			endpointName:  "",
+			errorExpected: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(_ *testing.T) {
+
+			req := &computeagent.AddNICInternalRequest{
+				NicID:        test.nicID,
+				EndpointName: test.endpointName,
+			}
+
+			_, err := agent.AddNIC(ctx, req)
+			if test.errorExpected && err == nil {
```

Comment:
if we know what the expected errors are, maybe we should also check them?

Reply:
We're just making sure that there's an error when the input is incorrect. If it was a lower down error, I think maybe it would be good to make sure we're erroring in the expected place, but not sure it's necessary here.

Reply:
fair enough.